### DISCUSSION TITLE
Create new associated models in a nested form

### DISCUSSION THREAD

**Discussion by iainbeeston** [avo-hq/avo#973] [Help]
Is there a way to create a new associated object from within the form of the parent? For example, if Bucket has many Holes, is there a way to create new Holes from the form for creating Buckets? Or do you have to create a Bucket, then create Holes one by one and attach them to the Bucket?

I'm not sure if this should be feature request or not, perhaps the feature already exists but I can't see it in the documentation, or perhaps it doesn't but there is a workaround?

**Comment by adrianthedev** [comment]
Hey @iainbeeston. At the moment it isn't possible using Avo's DSL. You could create a custom field and model it with the Hole inputs you need, submit that one and handle the Hole creation on your end, but that's probably not what you're looking for.

This feature [was requested](https://github.com/avo-hq/avo/discussions/779) before. It's not the easiest thing to implement in a way that benefits everybody in the same Avo way.

I'll add it to the roadmap for better visibility.

**Reply by iainbeeston** [reply]
Actually, I'm not sure I need this after all. If a bucket has many holes, and you define `field :holes, as: :has_many, show_on: :edit` it's very easy to add/remove nested holes.

**Reply by adrianthedev** [reply]
Oh, yeah. I forgot to mention that the edit screen can hold `has_many` associations.

Creating associations on the new screen introduces some major complexity and can lead to a lot of edge-cases.

I'll put this on the back-burner for now.

**Comment by adrianthedev** [comment]
I posted [this](https://docs.avohq.io/2.0/recipes/nested-records-when-creating.html) guide that shows an initial iteration on how nested records on create screen can be right now.
